### DISCUSSION TITLE
the scipy package shown on my webbrowser has different size as the one in the pyodide builded version

### DISCUSSION THREAD

**Discussion by homebovine** [pyodide/pyodide#4013] [Q&A]
I am currently use pyodide.js to dynamicly load scipy package in javascript from cdn. I found that the scipy package is 10M. But when I download pyodide-0.23.4.tar.bz2 from github and tried include pyodide locally. The size of scipy is 35M. What is the difference between pyodide-0.23.4.tar.bz2 and that on the cdn?

Also, I only used three functions in scipy. What is the best way to reduce the scipy size according to my own script? Could I manually delete some module in scipy inside .whl?

**Comment by ryanking13** [comment]
> I am currently use pyodide.js to dynamicly load scipy package in javascript from cdn. I found that the scipy package is 10M. But when I download pyodide-0.23.4.tar.bz2 from github and tried include pyodide locally. The size of scipy is 35M.

10M is brotli-compressed size of scipy wheel, while 35M is a real size of it. The CDN compresses the wheel. 

> Could I manually delete some module in scipy inside .whl?

Well I think it is possible, but I don't know how scipy submodules are tightly coupled, so I think you need to ask this to scipy maintainers.

**Reply by homebovine** [reply]
Thank you very much. I figured it out and successfully reduced scipy whl substantially based on my needs. I have some additional questions. The loading of repodata.json file is slow (~400ms on average). Is it file only used for Integrity check? If so, could I speed up the loading by using checkIntegrity = False when loading the package. Also, if I specify checkIntegrity = False, will pyodide perform check for the dependent packages. For example if I load scipy with checkIntegrity = False, will it check clapack's integrity.

**Reply by ryanking13** [reply]
> The loading of repodata.json file is slow (~400ms on average). Is it file only used for Integrity check?

repodata.json (it will be renamed to pyodide-lock.json in the next 0.24.0 release by the way), is a file that contains various information about a package including a dependency resolution information.

> If so, could I speed up the loading by using checkIntegrity = False when loading the package.

Yes, setting `checkIntegrity=False` will speed up the package loading as it will skip the package integrity check. TBH I am not sure how much it would affect the performance, and it might depend on a browser JS engine, as we just pass through that parameter into [`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/fetch). So I guess you need to do some profiling.

> Also, if I specify checkIntegrity = False, will pyodide perform check for the dependent packages. For example if I load scipy with checkIntegrity = False, will it check clapack's integrity.

It applies to all dependent packages that are loaded together. So if you load scipy with checkIntegrity=False, it will not check integrity of its dependencies like numpy, clapack.
